src/shared/install.c

```diff
@@ -68,6 +68,16 @@ typedef struct {
 
 static int unit_file_lookup_state(UnitFileScope scope, const LookupPaths *paths, const char *name, UnitFileState *ret);
 
+bool unit_type_no_alias(UnitType type) {
+        return IN_SET(type,
+                      UNIT_BUSNAME,
+                      UNIT_MOUNT,
+                      UNIT_AUTOMOUNT,
+                      UNIT_SWAP,
+                      UNIT_SLICE,
+                      UNIT_SCOPE);
+}
```

Comment:
The question is whether to blacklist known bad units (as is done currently), or to whitelist known good units. It does not matter on the pid1 side, because pid1 won't load a unit type it doesn't know anyway, but it could matter on the installation side. I though it's better to blacklist, but I guess new unit types are not added often enough for this to matter.